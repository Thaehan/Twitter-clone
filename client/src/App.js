import React, { useState } from 'react';
import { View, StyleSheet, StatusBar, TextInput, Button } from 'react-native';
import { Provider } from 'react-redux';

import RootNavigator from './navigation/RootNavigator';
import store from './redux/store';
import { createUser, getUser } from './api/user';

export default function App() {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const buttonHandle = () => {
    createUser(username, password)
      .then((res) => console.log(res.data))
      .catch((error) => console.log(error));
  };

  return (
        // <Provider store={store}>
    //   <RootNavigator />
    // </Provider>

    <View style={styles.container}>
      <StatusBar style="auto" />
      <TextInput
        style={styles.textInput}
        value={username}
        onChangeText={setUsername}
      />
      <TextInput
        value={password}
        onChangeText={setPassword}
        style={styles.textInput}
      />
      <Button title="Create user" onPress={buttonHandle} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  textInput: {
    height: 50,
    backgroundColor: 'red',
  },
});

