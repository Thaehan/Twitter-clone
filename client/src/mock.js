import { logo } from './constants/ImageAssets';

const tweets = {
  avatar: logo,
  fullname: 'DoDat',
  username: 'Thaehan',
  text: 'Troi oi cai quan que gi zay ba',
};


export { tweets };
