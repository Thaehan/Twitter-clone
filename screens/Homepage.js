import { View } from 'react-native';

import HelloWorld from '../components/HelloWorld';
import Counting from '../components/Counting';
import SearchFilter from '../components/SearchFilter';

export default function Homepage() {
  return (
    <View>
      <HelloWorld/>
      <Counting/>
      <SearchFilter/>
    </View>
  )
}