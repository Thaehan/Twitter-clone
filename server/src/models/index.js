const dbConfig = require('../config/config.js');

const mongoose = require('mongoose');
mongoose.Promise = global.Promise;
mongoose.set('useNewUrlParser', true);
mongoose.set('useFindAndModify', false);
mongoose.set('useCreateIndex', true);
const db = {};
db.mongoose = mongoose;
db.url = dbConfig.url;

db.tutorials = require('./tutorial.model')(mongoose);
db.users = require('./user.model')(mongoose);

module.exports = db;
