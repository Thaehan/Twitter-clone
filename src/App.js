import { View, Text } from 'react-native';
import React from 'react';
import RootNavigator from './navigation/RootNavigator';

export default function App() {
  return <RootNavigator />;
}
