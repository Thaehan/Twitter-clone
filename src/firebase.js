import { initializeApp } from 'firebase/app';
import { getAuth } from 'firebase/auth';
import { getFirestore } from 'firebase/firestore/lite';
import { getStorage } from 'firebase/storage/';

const config = {
  apiKey: 'AIzaSyDZD0dNa5P0RrkliqTVSQALpBOfNfFZOtc',
  authDomain: 'twitter-clone-5bfb8.firebaseapp.com',
  projectId: 'twitter-clone-5bfb8',
  storageBucket: 'twitter-clone-5bfb8.appspot.com',
  messagingSenderId: '411924634090',
  appId: '1:411924634090:web:90d88045e92680d2eaba93',
  measurementId: 'G-C6C5W57VVZ',
};

const app = initializeApp(config);
const auth = getAuth(app);
const db = getFirestore(app);
const storage = getStorage(app);

export { app, auth, db, storage };
