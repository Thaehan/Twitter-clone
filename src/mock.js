import { logo, avatar } from './constants/ImageAssets';
import avatar0 from './assets/avatar0.png'
import avatar1 from './assets/avatar1.png'
import avatar2 from './assets/avatar2.png'
import avatar3 from './assets/avatar3.png'
import avatar4 from './assets/avatar4.png'
import banner0 from './assets/banner0.png'
import banner1 from './assets/banner1.png'
import banner2 from './assets/banner2.png'
import banner3 from './assets/banner3.png'

const tweets = {
  "tweetId": "850006245121695744",
  "userPosted": "1",
  "textContent": "1/ Today we\u2019re sharing our vision for the future of the Twitter API platform! @Nowano #Loach!\nhttps://t.co/XweGngmxlP",
  "mediaContent": "assets/black",
  "dateCreated": "Thu Apr 06 15:24:15 +0000 2017",
  "referedPostId": "",
  "userMentioned": ["5"]
};
const tweetsList = [
  {
    "tweetId": "0",
    "userPosted": "0",
    "textContent": "Good morning!! How everyone doing today?",
    "mediaContent": "assets/black",
    "dateCreated": "Thu Apr 06 15:24:15 +0000 2017",
    "referedPostId": "",
    "userMentioned": ["2"]
  },
  {
    "tweetId": "1",
    "userPosted": "1",
    "textContent": "Layout direction specifies the direction in which children and text in a hierarchy should be laid out. Layout direction also affects what edge start and end refer to. By default, React Native lays out with LTR layout direction. In this mode start refers to left and end refers to",
    "mediaContent": "assets/black",
    "dateCreated": "Thu Apr 06 15:24:15 +0000 2017",
    "referedPostId": "",
    "userMentioned": []
  },
  {
    "tweetId": "2",
    "userPosted": "2",
    "textContent": "Any body show me a good nft place?",
    "mediaContent": "assets/black",
    "dateCreated": "Thu Apr 06 15:24:15 +0000 2017",
    "referedPostId": "",
    "userMentioned": []
  },
  {
    "tweetId": "3",
    "userPosted": "3",
    "textContent": ":D",
    "mediaContent": "assets/black",
    "dateCreated": "Thu Apr 06 15:24:15 +0000 2017",
    "referedPostId": "",
    "userMentioned": []
  }
]
const userDatabase = [
  {
    userId: "0",
    username: "Đỗ Đạc",
    handle: "thaehan1998",
    avatar: avatar0,
    banner: banner0,
    bio: "I am Đạt",
    following: [],
    liked: [],
  },
  {
    userId: "1",
    username: "Danh",
    handle: "wowyboi",
    avatar: avatar1,
    banner: banner1,
    bio: "I am Danh",
    following: [],
    liked: [],
  },
  {
    userId: "2",
    username: "Snoop Cat",
    handle: "snoppKitty",
    avatar: avatar2,
    banner: banner2,
    bio: "Smoke catweed everyday",
    following: [],
    liked: [],
  },
  {
    userId: "3",
    username: "Uyennhu",
    handle: "uwu",
    avatar: avatar3,
    banner: banner3,
    bio: "Sike fam",
    following: [],
    liked: [],
  },
]



const thaehan = {
  username: "Nick Ao Do Ven Dat",
  handle: "thaehan1998",
  password: "5",
  avatar: avatar,
  userId: "1"
}

export { tweets, thaehan, tweetsList, userDatabase };
