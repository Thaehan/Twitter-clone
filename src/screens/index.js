import Login from './Login';
import Signup from './Signup';
import Home from './Home';

export { Login, Signup, Home };
