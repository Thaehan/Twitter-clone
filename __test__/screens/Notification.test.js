import React from 'react';
// import renderer from 'react-test-renderer';\
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import { Notification } from '../../src/screens';
import store from '../../src/redux/store';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

jest.useFakeTimers();
jest.mock(
  'react-native/Libraries/Animated/NativeAnimatedHelper'
);

describe('<Notification />', () => {
  it('Render Notification', () => {
    const Stack = createNativeStackNavigator();
    const NotificationScreen = render(
      <Provider store={store}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen
              component={Notification}
              name={'Notification'}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </Provider>
    );
    expect(NotificationScreen.toJSON()).toMatchSnapshot();
  });
});
