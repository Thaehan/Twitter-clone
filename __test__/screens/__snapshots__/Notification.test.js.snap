// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> Render Notification 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Notification"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      >
        <RNSScreenStackHeaderSubview
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          type="left"
        >
          <View
            style={
              Object {
                "width": 60,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignContent": "center",
                  "alignSelf": "center",
                  "opacity": 1,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "",
                  }
                }
                style={
                  Object {
                    "borderRadius": 50,
                    "height": 30,
                    "width": 30,
                  }
                }
                testID="AvatarImage"
              />
            </View>
          </View>
        </RNSScreenStackHeaderSubview>
        <RNSScreenStackHeaderSubview
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          type="right"
        >
          <View
            style={
              Object {
                "width": 60,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignContent": "center",
                  "alignSelf": "center",
                  "borderRadius": 50,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreenStackHeaderSubview>
      </RNSScreenStackHeaderConfig>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "fontSize": 16,
              },
              Object {
                "backgroundColor": "white",
                "flex": 1,
                "width": "100%",
              },
            ]
          }
        >
          <RCTScrollView
            showsHorizontalScrollIndicator={false}
            showsVerticalScrollIndicator={false}
          >
            <View />
          </RCTScrollView>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
